sycl-fusion/passes/kernel-fusion/SYCLKernelFusion.cpp

```diff
@@ -438,10 +434,25 @@ Error SYCLKernelFusion::fuseKernel(
            "No jit_compiler::SYCLKernelInfo found");
     jit_compiler::SYCLKernelInfo &InputKernelInfo =
         *ModInfo->getKernelFor(FN.str());
-    appendKernelInfo(FusedKernelInfo, InputKernelInfo, UsedArgsMask);
+    appendKernelInfo(FusedParamKinds, FusedArgUsageMask, FusedAttributes,
+                     InputKernelInfo, UsedArgsMask);
     ++FuncIndex;
   }
 
+  // Add the information about the new kernel to the SYCLModuleInfo.
+  if (!ModInfo->hasKernelFor(FusedKernelName.str())) {
+    assert(FusedParamKinds.size() == FusedArgUsageMask.size());
+    jit_compiler::SYCLKernelInfo KI{FusedKernelName.str(),
+                                    FusedParamKinds.size()};
+    llvm::copy(FusedParamKinds, KI.Args.Kinds.begin());
+    llvm::copy(FusedArgUsageMask, KI.Args.UsageMask.begin());
+    KI.Attributes.insert(KI.Attributes.end(), FusedAttributes.begin(),
+                         FusedAttributes.end());
```

Comment:
```suggestion
    KI.Attributes.append(FusedAttributes.begin(), FusedAttributes.end());
```

Reply:
`KI.Attributes` is still an `std::vector` here, so we can't `append`. Also, this statement will become an `std::copy` into a `DynArray` in #12376 anyways.